tensorboard/webapp/routes/feature_flag_serializer.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Store} from '@ngrx/store';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+import {Location} from '../app_routing/location';
+import {SerializableQueryParams} from '../app_routing/types';
+import {State} from '../app_state';
+import * as selectors from '../selectors';
+import {EXPERIMENTAL_PLUGIN_QUERY_PARAM_KEY} from '../webapp_data_source/tb_feature_flag_data_source_types';
+import {FeatureFlagMetadata, FeatureFlagQueryParameters,} from '../webapp_data_source/tb_feature_flag_query_parameters';
+
+export function getFeatureFlagStates(store: Store<State>):
```

Comment:
Should this be a selector?

Reply:
This dependency on the store can actually just be removed after refactoring the `experimentalPlugins` logic!